### PR TITLE
Use shared/lib in build embeddings tool and run this tool as a module

### PR DESCRIPTION
**Pull request by shifucun**
This makes it possible to use shared libraries (more to come) and common config processing in NL server.

With this, more GCS download functions could be removed.

Also remove autogen input support since no autogen descriptions exist anymore.

### CODE REVIEW

File: tools/nl/embeddings/run_custom.sh [hunk 1]
```diff
@@ -15,10 +15,10 @@
...
 python3 -m venv .env
 source .env/bin/activate
-python3 -m pip install --upgrade pip
 pip3 install torch==2.2.2 --extra-index-url https://download.pytorch.org/whl/cpu
-pip3 install -r requirements.txt
+pip3 install -r tools/nl/embeddings/requirements.txt
 
-python3 build_custom_dc_embeddings.py "$@"
\ No newline at end of file
+python3 -m tools.nl.embeddings.build_custom_dc_embeddings "$@"
```
Comment (keyurva) [inline tools/nl/embeddings/run_custom.sh:24]: This script is not used anywhere. So we can delete it. If you do, please update the commands in the doc: https://github.com/datacommonsorg/website/blob/master/tools/nl/embeddings/build_custom_dc_embeddings.md

Also, can you run these commands as modules to verify that the functionality works as expected: https://github.com/datacommonsorg/website/pull/4216#discussion_r1595836596

Comment (shifucun) [reply tools/nl/embeddings/run_custom.sh:24]: Verified commands can run. I have also tested with local setup that talks to GCS storage.

File: server/routes/admin/html.py [hunk 2]
```diff
@@ -67,7 +67,8 @@ def load_data():
   # Build custom embeddings.
   command2 = [
       'python',
-      'build_custom_dc_embeddings.py',
+      '-m',
+      'tools.nl.embeddings.build_custom_dc_embeddings',
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/admin/html.py:71]: Updated and verified with docker build locally.

File: tools/nl/embeddings/build_custom_dc_embeddings.py [hunk 3]
```diff
@@ -11,27 +11,18 @@
...
 import yaml
 
-# Import gcs module from shared lib.
-# Since this tool is run standalone from this directory,
-# the shared lib directory needs to be appended to the sys path.
-_THIS_DIR = os.path.dirname(os.path.abspath(__file__))
-_SHARED_LIB_DIR = os.path.join(_THIS_DIR, "..", "..", "..", "shared", "lib")
-sys.path.append(_SHARED_LIB_DIR)
-import gcs  # type: ignore
+from shared.lib import gcs
```
[the remark this answers is not in the document]
Comment (shifucun) [reply tools/nl/embeddings/build_custom_dc_embeddings.py:22]: indeed!

File: tools/nl/embeddings/run_custom.sh (hunk 1 above)
Comment (shifucun) [reply tools/nl/embeddings/run_custom.sh:24]: I found the entire doc uses this script, so will keep as is. As I plan to consolidate the build embedding tool later, can remove at that time.

File: tools/nl/embeddings/build_embeddings.py [hunk 4]
```diff
@@ -180,23 +182,15 @@ def build(ctx, curated_input_dirs: List[str], local_merged_filepath: str,
...
     df_svs = pd.DataFrame()
 
-  # Append autogen CSVs if any.
-  autogen_dfs = []
-  for autogen_csv in sorted(glob.glob(autogen_input_filepattern)):
-    print(f'Processing autogen input file: {autogen_csv}')
-    autogen_dfs.append(pd.read_csv(autogen_csv).fillna(""))
-  if autogen_dfs:
-    df_svs = pd.concat([df_svs] + autogen_dfs)
-    df_svs = df_svs.drop_duplicates(subset=utils.DCID_COL)
```
Comment (pradh) [inline tools/nl/embeddings/build_embeddings.py:190]: If we're getting rid of autogen files support, should we also delete them?

**Review by pradh — APPROVED** [review]
Amazing cleanup!!

File: tools/nl/embeddings/build_embeddings.py (hunk 4 above)
Comment (shifucun) [reply tools/nl/embeddings/build_embeddings.py:190]: yes, they have been deleted already...

**Review by keyurva — APPROVED** [review]
Very cool!

**Comment by shifucun** [comment]
Thanks for review!

**Pull request merged**